app/Libraries/Database.php

```diff
@@ -72,84 +77,212 @@ public static function getInstance(array $credentials = null): object
+      return $stmt->fetchAll(PDO::FETCH_ASSOC);
+    } catch (\Throwable $th) {
+      throw $th;
+    }
+  }
+
+  /**
+   * Get tables for specified database
    *
-   * @param string $param
-   * @param string $value
-   * @param string $type
    * @return void
    */
-  public function bind(string $param, string $value, string $type = null): void
+  public function getTables()
   {
-    if (is_null($type)) {
-      switch (true) {
-        case is_int($value):
-          $type = PDO::PARAM_INT;
-          break;
-        case is_bool($value):
-          $type = PDO::PARAM_BOOL;
-          break;
-        case is_null($value):
-          $type = PDO::PARAM_NULL;
-          break;
-        default:
-          $type = PDO::PARAM_STR;
-      }
+    $db = Database::getInstance();
+
+    if (!Request::input('db_name')) {
+      return null;
     }
 
-    $this->stmt->bindValue($param, $value, $type);
+    try {
+      $tables = $db->getTablesFromServer(Request::input('db_name'));
+
+      $tables = array_map(function ($table) {
+        $table = array_values($table);
+        return array_pop($table);
+      }, $tables);
+
+      return $tables;
+    } catch (\Throwable $th) {
+      throw $th;
+    }
   }
 
   /**
-   * Execute the prepared statement
+   * Retrieve all Character Sets
    *
-   * @return bool
+   * @return array
    */
-  public function execute(): bool
+  public function getCharsets(): array
   {
-    return $this->stmt->execute();
+    $stmt = self::$pdo->query("SHOW CHARSET");
+    $stmt->execute();
+    return $stmt->fetchAll(PDO::FETCH_ASSOC);
   }
 
   /**
-   * Get result set as array of objects
+   * Retrieve all collations
    *
-   * @return bool
+   * @return array
    */
-  public function resultSet(): bool
+  public function getCollations(): array
   {
-    $this->execute();
-    return $this->stmt->fetchAll(PDO::FETCH_OBJ);
+    $stmt = self::$pdo->query("SHOW COLLATION;");
+    $stmt->execute();
+    return $stmt->fetchAll(PDO::FETCH_ASSOC);
   }
 
   /**
-   * Get single record as object
+   * Create new database on server
    *
-   * @return bool
+   * @param string $dbName
+   * @param string $charset
+   * @param string $collation
+   * @return void
    */
-  public function single(): bool
+  public function createDatabase(string $dbName, string $charset, string $collation): void
   {
-    $this->execute();
-    return $this->stmt->fetch(PDO::FETCH_OBJ);
+    $sql = "CREATE DATABASE $dbName CHARACTER SET $charset COLLATE $collation";
+
+    try {
+      self::$pdo->query($sql);
+    } catch (\Throwable $th) {
+      throw $th;
+    }
   }
 
   /**
-   * Get row count
+   * Retrieves table data
    *
-   * @return int
    */
-  public function rowCount(): int
+  public function fetchTableContents($database, $tableName)
+  {
+    $stmt = self::$pdo->query("USE $database;");
+    $stmt->execute();
+
+    try {
+      $stmt = self::$pdo->query("SELECT * FROM $tableName;");

```

Comment:
Same as above: $stmt -> $statement
Also, you are missing bindings here. `:databaseName` instead of `$databaseName` for security reasons.